geometric_kernels/lab_extras/jax/extras.py

```diff
@@ -38,3 +38,30 @@ def logspace(start: B.JAXNumeric, stop: _Numeric, num: int = 50):  # type: ignor
     Return numbers spaced evenly on a log scale.
     """
     return jnp.logspace(start, stop, num)
+
+
+@dispatch
+def degree(a: B.JAXNumeric):  # type: ignore
+    """
+    Diagonal matrix with x as main diagonal.
+    """
+    degrees = a.sum(axis=0)  # type: ignore
+    return jnp.diag(degrees)
+
+
+@dispatch
+def eigenpairs(L: B.JAXNumeric, k: int):
+    """
+    Obtain the k highest eigenpairs of a symmetric PSD matrix L.
+    """
+    l, u = jnp.linalg.eigh(L)
+    return l[:k], u[:, :k]
+
+
+@dispatch
+def set_value(a: B.JAXNumeric, index: B.Numeric, value: B.Numeric):
+    """
+    Set a[index] = value.
```

Comment:
what kind of shapes do you expect for `a`, `index`, and `value`?

Reply:
This function is just used to set the first eigenvalue to zero basically, so I'll change the signatures of index and value to `B.Number`. This way, if the function is called with any signature that's not `set_value(array, int, float)`, it'll fail.

Reply:
Resolved.